packages/dds/matrix/src/matrix.ts

```diff
@@ -250,11 +320,6 @@ export class SharedMatrix<T = any>
 				r++;
 			}
 		}
-
-		// Avoid reentrancy by raising change notifications after the op is queued.
-		for (const consumer of this.consumers.values()) {
-			consumer.cellsChanged(rowStart, colStart, rowCount, colCount, this);
```

Comment:
Consider preventing users from entering reentrancy (i.e. from these callback calling back into Matrix and changing it). While old code was also susceptible to this (in case of reentrancy updates would possibly fire for rows/columns that no longer exist or wrong indexex, if reentrant code did such changes), the new code is a bit more susceptible to this (this.setCellCore above in a loop could find itself operating with wrong indexex).

Something like that:

```ts
this.protectAgainstReentrancy(() => {
   ...
   this.setCellCore(r, c, value);
   ...
});

private this.protectAgainstReentrancy(callback: () => void) {
    assert(this.reentrantCount == 0);
    this.reentrantCount++;
    callback();
    this.reentrantCount--;
    assert(this.reentrantCount == 0);
}

```

Reply:
Actually such code could be within setCellCore itself - once for whole class, if that's the only place where we raise any events / change any structure. There might be more places to sprinkle it (like operations on rows/columns)